### ISSUE TITLE
[Bug] TargetSmart SmartMatch not showing up on documentation page

### ISSUE THREAD

**Issue by elyse-weiss** [move-coop/parsons#804]
## Detailed Description
noticed that we have TS SmartMatch in the code: https://github.com/move-coop/parsons/blob/main/parsons/targetsmart/targetsmart_smartmatch.py
but it’s not on the documentation page: https://move-coop.github.io/parsons/html/stable/targetsmart.html

**Comment by shaunagm** [comment]
Is this it? I'm not familiar with TargetSmart so I'm not sure if this actually grabs everything:

**Comment by elyse-weiss** [comment]
Yes! I think we can make the ask of TargetSmart to take the refactor on, if you can outline it clearly in an issue.

**Comment by shaunagm** [comment]
Most of our connectors have all of their code in one file. That makes it easy to include the docstrings in our documentation by linking to that single file in the docs. You can see a randomly chosen example, ActionKit, [here](https://github.com/move-coop/parsons/blob/main/docs/action_kit.rst#api).

Unusually, the TargetSmart connector is created through multiple inheritance from objects in several files. These each have their own docs page, which are linked out by the main docs page. This means the smartmatch documentation is technically findable, but confusing and difficult to find, as you experienced Elyse.

It's probably not worth refactoring the TargetSmart connector right now. There's tons of non-standard Parsons connectors and until we start prioritizing standardization it's just not the best use of dev time. 

But! We can definitely make the docs easier to understand. I'd recommend redoing the docs so all the documentation is on a single page - basically add https://github.com/move-coop/parsons/blob/main/docs/targetsmart_api.rst and https://github.com/move-coop/parsons/blob/main/docs/targetsmart_automation_workflows.rst to the main doc, https://github.com/move-coop/parsons/blob/main/docs/targetsmart.rst
